### ISSUE TITLE
Specify parameters in parallel_backend context manager

### ISSUE THREAD

**Issue by mrocklin** [joblib/joblib#322]
The current API to specify an external backend for joblib is, I think, the `paralell_backend` context manager:

```python
with parallel_backend('name-of-my-backend'):
    ...
```

Some backends may require parameters.  How should these be passed from user to backend?  For example, dask's `distributed` scheduler needs to know the location of the scheduler node, often a hostname/port pair.  What would be the right way to pass this in?

```python
with parallel_backend('distributed', scheduler='myserver:8786'):
    ...
```

One thought is that while strings are convenient for the common case, there should maybe be a way to pass in some full featured object.  This would also allow people to avoid the dance around registration

```python
with parallel_backend(my_parallel_object):
    ...
```

**Comment by lesteve** [comment]
> One thought is that while strings are convenient for the common case, there should maybe be a way to pass in some full featured object. This would also allow people to avoid the dance around registration

This seems reasonable. IIRC @ogrisel, when we discsussed that a little while ago, thought that passing an object to Parallel was already possible so I guess wouldn't object this feature.

**Comment by ogrisel** [comment]
Indeed, I wonder why I did not think about that in the first place :)

I will do a PR.

**Comment by ogrisel** [comment]
>  IIRC @ogrisel, when we discussed a little while ago, thought that passing an object into the Parallel backend parameter was already possible so I guess wouldn't object this feature.

Yes but passing instances rendered the code more complex so I decided not to do it for now. I will do my PR on top of the branch in #321.

**Comment by ogrisel** [comment]
I merged #325, thanks for the feedback @mrocklin.
